### PR TITLE
Book model

### PR DESCRIPTION
**Pull request by wleoguty91**
## Summary

- Added book model

## Trello Card

- https://trello.com/c/2nKdmq4F/14-crear-modelo-de-book

Screenshot from 2019-12-06 12-47-23

### CODE REVIEW

File: src/main/java/wolox/training/models/Book.java [hunk 1]
```diff
@@ -0,0 +1,124 @@
+package wolox.training.models;
+
+import javax.persistence.Column;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+
```
Comment (facucompe) [inline src/main/java/wolox/training/models/Book.java:10]: Missing
```
@Entity
@Table(name="book")
```

File: src/main/java/wolox/training/models/Book.java [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+public class Book {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    private long id;
+
+    @Column()
```
Comment (facucompe) [inline src/main/java/wolox/training/models/Book.java]: I think you can remove those parentheses

File: src/main/java/wolox/training/models/Book.java [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+    @Column(nullable = false)
+    private String year;
+
+    @Column(nullable = false)
+    private Integer pages;
+
+    @Column(nullable = false)
+    private String isbn;
+
+    public Book(String author, String image, String title, String subtitle,
```
Comment (facucompe) [inline src/main/java/wolox/training/models/Book.java]: You need to add a constructor with no args too. JPA use it to create the instance object.

File: src/main/java/wolox/training/models/Book.java [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+        setImage(image);
+        setTitle(title);
+        setSubtitle(subtitle);
+        setPublisher(publisher);
+        setYear(year);
+        setPages(pages);
+        setIsbn(isbn);
+    }
+
+    public String getGenre() {
```
Comment (facucompe) [inline src/main/java/wolox/training/models/Book.java]: I recommend using [lombok](https://projectlombok.org/setup/gradle) to implement getters and setters

**Review by facucompe — CHANGES_REQUESTED** [review]
Check the database to see if JPA is mapping the model correctly

File: src/main/java/wolox/training/models/Book.java (hunk 1 above)
Comment (leonplondon) [reply src/main/java/wolox/training/models/Book.java:10]: Also add appropriate indentation

**Review by facucompe — APPROVED** [review]

**Review by leonplondon — APPROVED** [review]

**Pull request merged**